### PR TITLE
(MODULES-9319) Align Module

### PR DESCRIPTION
**Pull request by michaeltlombardi**
- [x] pdk update
- [x] Rubocop
- [x] Unit tests validation
- [x] Acceptance tests validation
- [x] Update Puppet minimum version
- [x] Module Contribution Guide

### CODE REVIEW

File: spec/lib/dsc_utils.rb [hunk 1]
```diff
@@ -43,75 +43,75 @@ def get_dsc_vendor_resource_abs_path(dsc_module, expect_ps_failure)
...
+      # block with arity of 1 or greater, hand back the result object
+    else
+      yield result
     end
   end
   result
 end
 
-def assert_dsc_resource(dsc_resource_type, dsc_module, expect_ps_failure = false, expect_dsc_failure = false, dsc_properties)
+def assert_dsc_resource(dsc_resource_type, dsc_module, dsc_properties, expect_ps_failure = false, expect_dsc_failure = false)
```
Comment (tphoney) [inline spec/lib/dsc_utils.rb]: the ordering of this has changed, is it intentional

Comment (tphoney) [reply spec/lib/dsc_utils.rb]: it looks like it was changed in other places too, so i'll assume it is intentional

Comment (michaeltlombardi) [reply spec/lib/dsc_utils.rb]: I remember the linting failing if I didn't move this around, but it's distant enough I don't directly remember exactly now. :(

File: build/dsc/resource.rb [hunk 2]
```diff
@@ -138,7 +140,7 @@ def valid_ensure_present_values
         elsif ensure_property.nil?
           ['present']
         # a non-nil list of ensure values with 'absent' filtered
-        elsif (present = ensure_property.values.collect { |v| v.downcase }.select { |v| v != 'absent' })
+        elsif (present = ensure_property.values.map { |v| v.downcase }.reject { |v| v == 'absent' })
```
[the remark this answers is not in the document]
Comment (michaeltlombardi) [reply build/dsc/resource.rb]: @ThoughtCrhyme I only changed the code, especially in the builder, as much as necessary to pass rubocop.

[the remark this answers is not in the document]
Comment (ThoughtCrhyme) [reply build/dsc/resource.rb]: word

**Pull request merged**